src/H5VLint.c

```diff
@@ -1355,12 +1351,12 @@ H5VL__register_connector_by_value(H5VL_class_value_t value, hbool_t app_ref, hid
     FUNC_ENTER_PACKAGE
 
     /* Set up op data for iteration */
-    op_data.kind     = H5VL_GET_CONNECTOR_BY_VALUE;
-    op_data.u.value  = value;
-    op_data.found_id = H5I_INVALID_HID;
+    op_data.key.kind    = H5VL_GET_CONNECTOR_BY_VALUE;
+    op_data.key.u.value = value;
+    op_data.found_id    = H5I_INVALID_HID;
 
     /* Check if connector is already registered */
-    if (H5I_iterate(H5I_VOL, H5VL__get_connector_cb, &op_data, TRUE) < 0)
+    if (H5I_iterate(H5I_VOL, H5VL__get_connector_cb, &op_data, app_ref) < 0)
```

Comment:
@derobins I believe this is the correct change from looking at the similar H5VL__register_connector_by_name, but if the two are supposed to act differently from each other, I'll change this back.